### PR TITLE
chore: add more options to ManifestOptions

### PR DESCRIPTION
**Pull request by rehhouari**
Added some more missing types for ManifestOptions and and a catch-all at the end in case some new ones are added that this library don't document

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -147,4 +147,52 @@ export interface ManifestOptions {
...
+   */
+  related_applications: {
+    platform: string,
+    url: string,
+    id?: string,
+  }[]
+  /**
+   * @default false
+   */
+  prefer_related_applications: boolean
```
Comment (userquin) [inline src/types.ts:161]: should be `prefer_related_applications?: boolean` ([spec](https://developer.mozilla.org/en-US/docs/Web/Manifest/prefer_related_applications))

File: src/types.ts [hunk 2]
```diff
@@ -147,4 +147,52 @@ export interface ManifestOptions {
    * @default A combination of `routerBase` and `options.build.publicPath`
    */
   publicPath: string
+  /**
+   * @default []
+   */
+  related_applications: {
```
Comment (userquin) [inline src/types.ts:153]: should be `related_applications?: {...}[]` [spec](https://developer.mozilla.org/en-US/docs/Web/Manifest/prefer_related_applications)

**Comment by userquin** [comment]
@rehhouari please review mandatory entries, all entries you added are no mandatory.

The `catch-all` entry should not be added, since it can be anything in the future, I'll prefer add them when necessary and have them typed, since these entries shouldn't change along time.

**Comment by userquin** [comment]
@rehhouari or add defaults to `options.ts` module

imagen

File: src/types.ts (hunk 1 above)
Comment (rehhouari) [reply src/types.ts:161]: yes true sorry!

**Comment by rehhouari** [comment]
Like this?

**Comment by userquin** [comment]
@rehhouari just remove `catch-all`, keep original `options.ts`, there is no need to add those defaults, and sorry for the inconveniences. I forgot we are using `Partial` and so there is no need to include defaults for those guys.

And thx for the PR.

**Comment by rehhouari** [comment]
done ^^

**Review by userquin — APPROVED** [review]

**Pull request merged**